Shared/Samples/Run valve isolation trace/RunValveIsolationTraceView.Model.swift

```diff
@@ -0,0 +1,402 @@
+            let terminalName = "Load"
+            let globalID = UUID(uuidString: "98A06E95-70BE-43E7-91B7-E34C9D3CB9FF")!
+            
+            // Create a default starting location.
+            if let networkSource = utilityNetwork.definition?.networkSource(named: networkSourceName),
+               let assetType = networkSource.assetGroup(named: assetGroupName)?.assetType(named: assetTypeName),
+               let startingLocation = utilityNetwork.makeElement(assetType: assetType, globalID: globalID) {
+                // Set the terminal for the location. (For our case, use the "Load" terminal.)
+                startingLocation.terminal = assetType.terminalConfiguration?.terminals.first(where: { $0.name == terminalName })
+                return startingLocation
+            } else {
+                return nil
+            }
+        }
+        
+        /// Adds a graphic to the graphics overlay.
+        /// - Parameters:
+        ///   - location: The `Point` location to place the graphic..
+        ///   - traceLocationType: The textual description of the trace location type.
+        private func addGraphic(for location: Point, traceLocationType: String) {
+            let graphic = Graphic(
+                geometry: location,
+                attributes: ["TraceLocationType": traceLocationType]
+            )
+            parametersOverlay.addGraphic(graphic)
+        }
+        
+        /// Sets the selected filter barrier category and updates the status text.
+        func selectCategory(_ category: UtilityCategory) {
+            selectedCategory = category
+            statusText = "\(category.name) selected."
+            traceEnabled = true
+        }
+        
+        /// Unselects the selected filter barrier category and updates the status text.
+        func unselectCategory(_ category: UtilityCategory) {
+            selectedCategory = nil
+            traceEnabled = hasFilterBarriers
+            statusText = "Tap on the map to add filter barriers, or run the trace directly without filter barriers."
+        }
+        
+        /// Runs a trace with the pending trace configuration and selects features in the map that
+        /// correspond to the element results.
+        @MainActor
+        func trace() async {
+            // Clear previous trace results.
+            layers.forEach { $0.clearSelection() }
+            
+            tracingActivity = .runningTrace
+            traceEnabled = false
+            
+            let configuration = makeTraceConfiguration(category: selectedCategory)
+            traceParameters.traceConfiguration = configuration
+            let traceResults: [UtilityElementTraceResult]
+            do {
+                traceResults = try await utilityNetwork
+                    .trace(using: traceParameters)
+                    .compactMap { $0 as? UtilityElementTraceResult }
+            } catch {
+                statusText = "Trace failed."
```

Comment:
I'm not sure if `tracingActivity` can cover both the tracing states and status text, since tracing status is displayed in an overlay with an activity indicator and the status text is displayed in a banner style overlay.

Reply:
Yeah, my comment is not applicable. Let me find another way to organize these status text… 🤕